src/types.ts

```diff
@@ -5,9 +5,68 @@ import type { SegmentAnalytics } from './extensions/segment-integration'
 export type Property = any
 export type Properties = Record<string, Property>
 
+export const COPY_AUTOCAPTURE_EVENT = '$copy_autocapture'
+
+export const knownUnEditableEvent = [
+    '$web_experiment_applied',
+    '$$client_ingestion_warning',
+    '$feature_enrollment_update',
+    '$feature_flag_called',
+    'survey dismissed',
+    'survey sent',
+    'survey shown',
+    '$snapshot',
+] as const
+
+/**
+ * These events are not processed by the `beforeCapture` function
+ */
+export type KnownUnEditableEvent = typeof knownUnEditableEvent[number]
+
+export const knownUnsafeEditableEvent = [
+    '$pageview',
+    '$pageleave',
+    '$identify',
+    '$set',
+    '$groupidentify',
+    '$create_alias',
```

Comment:
I wonder if those should be `knownUnEditableEvent` since they are called max 1 time per session or if something changes, usually isn't a huge amount of events, and missing them would cause "I don't find my users" type of support tickets.